### PR TITLE
Add sample implementing a login with github oauth service

### PR DESCRIPTION
**Pull request by gtardif**
This illustrate the login flow that can be implemented in extensions with extension SDK version 0.3.3 and above (starting with Docker Desktop 4.16)

Signed-off-by: Guillaume Tardif <guillaume.tardif@gmail.com>

### CODE REVIEW

File: samples/oauth-sample/ui/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import Button from "@mui/material/Button";
+import { createDockerDesktopClient } from "@docker/extension-api-client";
+import { Stack, TextField, Typography } from "@mui/material";
+
+const client = createDockerDesktopClient();
+
+const client_id = "xxxx";
+const client_secret = "xxxx";
```
Comment (benja-M-1) [inline samples/oauth-sample/ui/src/App.tsx:9]: ** suggestion:** Could you add instructions in the readme so that the sample works out of the box?

**Review by benja-M-1 — COMMENTED** [review]
That is so cool to have an example for this feature!

File: samples/oauth-sample/ui/src/App.tsx (hunk 1 above)
Comment (gtardif) [reply samples/oauth-sample/ui/src/App.tsx:9]: Added in the readme

File: samples/oauth-sample/ui/src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    ddClient.host.openExternal(
+      `https://github.com/login/oauth/authorize?client_id=${client_id}`
+    );
+  };
+
+  return (
+    <>
+      <Typography variant="h3">Docker extension Oauth demo</Typography>
+      <Typography variant="body1" color="text.secondary" sx={{ mt: 2 }}>
+        This is a basic page using Oauth to login in using github Oauth as an
```
Comment (benja-M-1) [inline samples/oauth-sample/ui/src/App.tsx:51]: nit: 
```suggestion
        This is a basic page using OAuth to login in using GitHub OAuth as an
```

File: samples/oauth-sample/ui/src/App.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        example.
+      </Typography>
+      <Typography variant="body1" color="text.secondary" sx={{ mt: 2 }}>
+        Pressing the below button will trigger a login flow and retrieve a
+        github authentication token once the user is authenticated.
```
Comment (benja-M-1) [inline samples/oauth-sample/ui/src/App.tsx:56]:
```suggestion
        GitHub authentication token once the user is authenticated.
```

**Review by benja-M-1 — APPROVED** [review]
I made some minor suggestions, mostly nitpicking.

**Review by benja-M-1 — APPROVED** [review]

**Pull request merged**